File: src/main/kotlin/ch/fdlo/hoerbuchspion/webservice/data/AlbumDetails.java
package ch.fdlo.hoerbuchspion.webservice.data;

import javax.persistence.Embeddable;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;

@Embeddable
public class AlbumDetails {
  private int totalTracks;
  private long totalDurationMs;
  private boolean allTracksNotExplicit = true;
  private boolean allTracksPlayable = true;
  private String previewURL = "";
  private int popularity;
  private String label = "";
  private String copyright = "";
  @Enumerated(EnumType.STRING)
  private Language assumedLanguage = Language.UNKNOWN;

  // Required by JPA
  private AlbumDetails() {}

  public void setPopularity(int popularity) {
    this.popularity = popularity;
  }

  public void setLabel(String label) {
    this.label = label;
  }

  public void setCopyright(String copyright) {
    this.copyright = copyright;
  }

  public void setAssumedLanguage(Language assumedLanguage) {
    this.assumedLanguage = assumedLanguage;
  }

  public enum Language {
    DE, EN, FR, ES, IT, UNKNOWN;
  }
}

File: src/main/kotlin/ch/fdlo/hoerbuchspion/webservice/data/AlbumDetails.kt
package ch.fdlo.hoerbuchspion.webservice.data

import javax.persistence.Embeddable
import javax.persistence.EnumType
import javax.persistence.Enumerated

@Embeddable
class AlbumDetails
private constructor() { // We need a default constructor because of JPA
    val totalTracks = 0
    val totalDurationMs: Long = 0
    val allTracksNotExplicit = true
    val allTracksPlayable = true
    val preview = "" // URL to preview clip
    // TODO: Check why 'val' and 'var' are mixed
    var popularity = 0
    var label = ""
    var copyright = ""

    @Enumerated(EnumType.STRING)
    var assumedLanguage = Language.UNKNOWN

    enum class Language {
        DE, EN, FR, ES, IT, UNKNOWN;

        companion object {
            fun fromISO_639_1(lang: String): Language {
                return when (lang) {
                    "de" -> DE
                    "en" -> EN
                    "fr" -> FR
                    "es" -> ES
                    "it" -> IT
                    else -> UNKNOWN
                }
            }
        }
    }
}